acceptance/suites/pre_suite/upgrade/10_install_puppet.rb

```diff
@@ -0,0 +1,31 @@
+step "Install PC1 repository" do
+  install_puppetlabs_release_repo(hosts, 'pc1')
+end
+
+step "Install legacy Puppet agents" do
+  default_puppet_version = '1.10.1'
+  puppet_version = ENV['PUPPET_LEGACY_VERSION']
+  if not puppet_version
+    logger.info "PUPPET_LEGACY_VERSION is not set!"
+    logger.info "  using default value of #{default_puppet_version}"
+    puppet_version = default_puppet_version
+  end
+
+  hosts.each { |host| host.install_package('puppet-agent', '', puppet_version) }
+end
+
+step "Install legacy Puppetserver" do
+  master.install_package('puppetserver', '', '2.7.2')
```

Comment:
For the PC1 repo config as the base, that might also be beneficial for avoiding having to pin either a puppet-agent or puppetserver version to install, which could be nice.